pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -390,45 +372,110 @@ func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Contex
 		log.Error(msg)
 		return errors.New(msg)
 	}
-	if len(clientVms) == 0 {
-		// Prepare the CnsVolumeACLConfigureSpec
-		cnsVolumeID := cnstypes.CnsVolumeId{
-			Id: volumeID,
+	if !delete && len(clientVms) == 0 {
```

Comment:
How about changing this to `if !delete {` and moving `if len(clientVMs) == 0` into that control path. That way you can bring `AddClientVMToIPList` on line 401 into this path too. Easier to read IMO..

Reply:
Addressed in the latest commit